filebeat/harvester/reader/encoder.go

```diff
@@ -0,0 +1,35 @@
+package reader
+
+import (
+	"io"
+	"time"
+
+	"github.com/elastic/beats/filebeat/harvester/encoding"
+)
+
+// Encode reader produces lines by reading lines from an io.Reader
+// through a decoder converting the reader it's encoding to utf-8.
+type Encode struct {
```

Comment:
Ah right, we pass the codec, so it could be anything not only uft-8 if configured differently. In this case I would stick with `reader.Encode`?